### PR TITLE
[CARBONDATA-2801]Added documentation for flat folder

### PR DESCRIPTION
**Pull request by ravipesala**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6064/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6065/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6354/

File: docs/data-management-on-carbondata.md [hunk 1]
```diff
@@ -284,6 +286,20 @@ This tutorial is going to introduce all commands and data operations on CarbonDa
 	 ALTER TABLE employee SET TBLPROPERTIES (‘CACHE_LEVEL’=’Blocklet’)
 	 ```
 	 
+	 - **Support Flat folder**
+	 
+	  This feature allows all carbondata and carbonindex files to keep directy under tablepath. Currently all carbondata/carbonindex files written under tablepath/Fact/Part0/Segment_NUM folder and it is not same as hive/parquet folder structure. This feature makes all files written will be directly under tablepath, it does not maintain any segment folder structure.
```
Comment (jackylk) [inline docs/data-management-on-carbondata.md]: rephase to  `This feature allows all data files and index files to keep directy under tablepath`

Comment (jackylk) [reply docs/data-management-on-carbondata.md]: Can you also explain why this is needed, or what scenario user should consider to use this feature.

Comment (ravipesala) [reply docs/data-management-on-carbondata.md]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7663/

File: docs/data-management-on-carbondata.md [hunk 2]
```diff
@@ -284,6 +286,20 @@ This tutorial is going to introduce all commands and data operations on CarbonDa
 	 ALTER TABLE employee SET TBLPROPERTIES (‘CACHE_LEVEL’=’Blocklet’)
 	 ```
 	 
+	 - **Support Flat folder**
```
Comment (chenliang613) [inline docs/data-management-on-carbondata.md]: change to : 

**Support Flat folder same as Hive/Parquet**

Comment (ravipesala) [reply docs/data-management-on-carbondata.md]: ok

**Comment by chenliang613** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6399/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6079/

**Pull request closed without merging**